### ISSUE TITLE
Only export enum type

### ISSUE THREAD

**Issue by TomasBarry** [dotansimha/graphql-code-generator#3475]
**Is your feature request related to a problem? Please describe.**

The flexibility of GraphQL queries severely limits the use of exported types in the generated file of Flow types.

Consider the following schema:

```es6
/* @flow */

/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {|
  ID: string,
  String: string,
  Boolean: boolean,
  Int: number,
  Float: number,
  ISO8601Date: any,
  ISO8601DateTime: any,
|};

export const MyEnumValues = Object.freeze({
  A: 'a', 
  B: 'b', 
  C: 'c'
});

export type MyEnum = $Values<typeof MyEnumValues>;

export type MyType = {|
  id: $ElementType<Scalars, 'ID'>,
  field: MyEnum,
|};
```

It would be fragile to type the return value of a GraphQL query with the type of `MyType` as Flow would not be able to assist us if we changed the query from:

```graphql
query {
  myType {
    id,
    field
  }
}
```

To:

```graphql
query {
  myType {
    field
  }
}
```

**Describe the solution you'd like**

We would like a mechanism to just generate types for enums and skip over GraphQL types.

**Describe alternatives you've considered**

The alternative is manually defining and maintaining enums in Flow or limiting ourselves to just importing the enums. However, if the types can be imported it is not reasonable for us to be able to police every commit from using them.

**Comment by dotansimha** [comment]
@TomasBarry 
The basic `flow` plugin only generates the base types, based on your GraphQL Schema. 
If you wish to get type safety based on your queries, you should use `flow-operations` as well, and use `documents:` section to specify your GraphQL operations. 
It will generate Flow types based on your selection set, and if you change your query if will generate new types according to the fields you are asking for. 

Does that covers you use-case? I don't understand why generating only enums is related?

**Comment by dotansimha** [comment]
Closing. @TomasBarry let me know if it's still relevant.

**Comment by darkbasic** [comment]
@dotansimha the feature could make sense for another use case. Let's say that you want to use a `.d.ts` declaration file instead of a `.ts` one, but you also **don't** want to use `enumsAsTypes`: in that situation it could be useful to generate an additional `.ts` file with only the enums, so you can import them through your application while maintaining your declaration file.

**Comment by dotansimha** [comment]
@darkbasic I see. I'm not sure if it's related, but does const enum assertions works in `d.ts` files? (see https://github.com/dotansimha/graphql-code-generator/pull/3569) because this might help with that situations. 
It shouldn't be too complicated to create a small plugin that only generates enums, and it can share the same logic as the existing plugins... PRs are welcome ;)

**Comment by darkbasic** [comment]
@dotansimha that's the problem: const enums don't work in `.d.ts` files either.

**Comment by dotansimha** [comment]
Done in: https://github.com/dotansimha/graphql-code-generator/pull/3760

**Comment by TomasBarry** [comment]
Thank you @dotansimha!

**Comment by dotansimha** [comment]
Fixed in v1.13.4 :) 

You can now use:

```yaml
config:
  preResolveTypes: true
  onlyOperationTypes: true
```

And you'll get minimal types generated.

**Comment by dotansimha** [comment]
> @dotansimha thanks for putting in the work to add those additional flags! One issue is I would like to only generate enums and not create types in the .ts file. Is that possible?

It's not available at the moment, you can create a PR if you need that ;)

**Comment by Borduhh** [comment]
Took a whack at it quick based on https://github.com/dotansimha/graphql-code-generator/pull/3760/files#diff-59230f39cf86a6a2257155cba47d7c642aadb8697d3a925288a9c00093ac71d1 but it's a little more involved because of how scalars are built I think.

Even when updating the constructor to pass an empty object as scalars, it still outputs types:

**packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts**
```ts
constructor(
    protected _schema: GraphQLSchema,
    rawConfig: TRawConfig,
    additionalConfig: TPluginConfig,
    defaultScalars: NormalizedScalarsMap = DEFAULT_SCALARS
  ) {
    super(rawConfig, {
      enumPrefix: getConfigValue(rawConfig.enumPrefix, true),
      onlyEnums: getConfigValue(rawConfig.onlyEnums, false),
      onlyOperationTypes: getConfigValue(rawConfig.onlyOperationTypes, false),
      addUnderscoreToArgsType: getConfigValue(rawConfig.addUnderscoreToArgsType, false),
      enumValues: parseEnumValues({
        schema: _schema,
        mapOrStr: rawConfig.enumValues,
        ignoreEnumValuesFromSchema: rawConfig.ignoreEnumValuesFromSchema,
      }),
      declarationKind: normalizeDeclarationKind(rawConfig.declarationKind),
      // scalars: buildScalarsFromConfig(_schema, rawConfig, defaultScalars),
      fieldWrapperValue: getConfigValue(rawConfig.fieldWrapperValue, 'T'),
      wrapFieldDefinitions: getConfigValue(rawConfig.wrapFieldDefinitions, false),
      entireFieldWrapperValue: getConfigValue(rawConfig.entireFieldWrapperValue, 'T'),
      wrapEntireDefinitions: getConfigValue(rawConfig.wrapEntireFieldDefinitions, false),
      ignoreEnumValuesFromSchema: getConfigValue(rawConfig.ignoreEnumValuesFromSchema, false),
      directiveArgumentAndInputFieldMappings: transformDirectiveArgumentAndInputFieldMappings(
        rawConfig.directiveArgumentAndInputFieldMappings ?? {},
        rawConfig.directiveArgumentAndInputFieldMappingTypeSuffix
      ),
      ...additionalConfig,
      scalars: {},
    });
```

**dev-test/githunt/types.onlyEnums.ts**
```ts
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = {
  [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> &
  { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> &
  { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {};

/** A list of options for the sort order of the feed */
export enum FeedType {
  /** Sort by a combination of freshness and score, using Reddit's algorithm */
  Hot = "HOT",
  /** Newest entries first */
  New = "NEW",
  /** Highest score entries first */
  Top = "TOP"
}

/** The type of vote to record, when submitting a vote */
export enum VoteType {
  Cancel = "CANCEL",
  Down = "DOWN",
  Up = "UP"
}

export type OnCommentAddedSubscriptionVariables = Exact<{
  repoFullName: String;
}>;

export type OnCommentAddedSubscription = {
  __typename?: "Subscription";
  commentAdded?: {
    __typename?: "Comment";
    id: Int;
    createdAt: Float;
    content: String;
    postedBy: { __typename?: "User"; login: String; html_url: String };
  } | null;
};

export type CommentQueryVariables = Exact<{
  repoFullName: String;
  limit?: InputMaybe<Int>;
  offset?: InputMaybe<Int>;
}>;

export type CommentQuery = {
  __typename?: "Query";
  currentUser?: { __typename?: "User"; login: String; html_url: String } | null;
  entry?: {
    __typename?: "Entry";
    id: Int;
    createdAt: Float;
    commentCount: Int;
    postedBy: { __typename?: "User"; login: String; html_url: String };
    comments: Array<{
      __typename?: "Comment";
      id: Int;
      createdAt: Float;
      content: String;
      postedBy: { __typename?: "User"; login: String; html_url: String };
    } | null>;
    repository: {
      __typename?: "Repository";
      description?: String | null;
      open_issues_count?: Int | null;
      stargazers_count: Int;
      full_name: String;
      html_url: String;
    };
  } | null;
};

export type CommentsPageCommentFragment = {
  __typename?: "Comment";
  id: Int;
  createdAt: Float;
  content: String;
  postedBy: { __typename?: "User"; login: String; html_url: String };
};

export type CurrentUserForProfileQueryVariables = Exact<{
  [key: string]: never;
}>;

export type CurrentUserForProfileQuery = {
  __typename?: "Query";
  currentUser?: {
    __typename?: "User";
    login: String;
    avatar_url: String;
  } | null;
};

export type FeedEntryFragment = {
  __typename?: "Entry";
  id: Int;
  commentCount: Int;
  score: Int;
  createdAt: Float;
  repository: {
    __typename?: "Repository";
    full_name: String;
    html_url: String;
    description?: String | null;
    stargazers_count: Int;
    open_issues_count?: Int | null;
    owner?: { __typename?: "User"; avatar_url: String } | null;
  };
  vote: { __typename?: "Vote"; vote_value: Int };
  postedBy: { __typename?: "User"; html_url: String; login: String };
};

export type FeedQueryVariables = Exact<{
  type: FeedType;
  offset?: InputMaybe<Int>;
  limit?: InputMaybe<Int>;
}>;

export type FeedQuery = {
  __typename?: "Query";
  currentUser?: { __typename?: "User"; login: String } | null;
  feed?: Array<{
    __typename?: "Entry";
    id: Int;
    commentCount: Int;
    score: Int;
    createdAt: Float;
    repository: {
      __typename?: "Repository";
      full_name: String;
      html_url: String;
      description?: String | null;
      stargazers_count: Int;
      open_issues_count?: Int | null;
      owner?: { __typename?: "User"; avatar_url: String } | null;
    };
    vote: { __typename?: "Vote"; vote_value: Int };
    postedBy: { __typename?: "User"; html_url: String; login: String };
  } | null> | null;
};

export type SubmitRepositoryMutationVariables = Exact<{
  repoFullName: String;
}>;

export type SubmitRepositoryMutation = {
  __typename?: "Mutation";
  submitRepository?: { __typename?: "Entry"; createdAt: Float } | null;
};

export type RepoInfoFragment = {
  __typename?: "Entry";
  createdAt: Float;
  repository: {
    __typename?: "Repository";
    description?: String | null;
    stargazers_count: Int;
    open_issues_count?: Int | null;
  };
  postedBy: { __typename?: "User"; html_url: String; login: String };
};

export type SubmitCommentMutationVariables = Exact<{
  repoFullName: String;
  commentContent: String;
}>;

export type SubmitCommentMutation = {
  __typename?: "Mutation";
  submitComment?: {
    __typename?: "Comment";
    id: Int;
    createdAt: Float;
    content: String;
    postedBy: { __typename?: "User"; login: String; html_url: String };
  } | null;
};

export type VoteButtonsFragment = {
  __typename?: "Entry";
  score: Int;
  vote: { __typename?: "Vote"; vote_value: Int };
};

export type VoteMutationVariables = Exact<{
  repoFullName: String;
  type: VoteType;
}>;

export type VoteMutation = {
  __typename?: "Mutation";
  vote?: {
    __typename?: "Entry";
    score: Int;
    id: Int;
    vote: { __typename?: "Vote"; vote_value: Int };
  } | null;
};
```

Happy to take another whack with some guidance.

**Comment by Borduhh** [comment]
Solved this in #7718
